src/transactions.rs

```diff
@@ -0,0 +1,456 @@
+        pending_uids: Vec<u64>,
+    ) -> Result<(), KeystoreError<L>> {
+        for uid in pending_uids {
+            if let Some(txn_uid) = self.uids_awaiting_memos.remove(&uid) {
+                let txn_editor = self.get_mut(&txn_uid)?;
+                txn_editor.remove_pending_uid(uid).save()?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Remove a transaction from the pending index when it
+    pub async fn remove_pending(
+        &mut self,
+        timeout: u64,
+        uid: &TransactionUID<L>,
+    ) -> Result<(), KeystoreError<L>> {
+        if let Some(expiring) = self.expiring_txns.get_mut(&timeout) {
+            expiring.remove(uid);
+            if expiring.is_empty() {
+                self.expiring_txns.remove(&timeout);
+            }
+        }
+        Ok(())
+    }
+
+    /// Create an Transaction.
+    ///
+    /// When we first create a transaction it won't know all of it's values until
+    /// specific events happen on chain.  When the block for this transaction commited
+    /// We will learn it's Hash, Recept, and PendingUIDs as well as be certain of the
+    /// Fee and Asset Change records.
+    ///
+    /// Returns the editor for the created asset.
+    #[allow(clippy::too_many_arguments)]
+    pub fn create(
+        &mut self,
+        uid: TransactionUID<L>,
+        timeout: u64,
+        accounts: Vec<UserAddress>,
+        memos: Vec<Option<ReceiverMemo>>,
+        sig: Signature,
+        freeze_outputs: Vec<RecordOpening>,
+        inputs: Vec<RecordOpening>,
+        outputs: Vec<RecordOpening>,
+        time: DateTime<Local>,
+        asset: AssetCode,
+        kind: TransactionKind<L>,
+        senders: Vec<UserAddress>,
+        receivers: Vec<(UserAddress, RecordAmount)>,
+    ) -> Result<TransactionEditor<'_, L>, KeystoreError<L>> {
+        let txn = Transaction::<L> {
+            uid,
+            timeout,
+            hash: None,
+            pending_uids: HashSet::new(), //pending_uids
+            accounts,
+            memos,
+            sig,
+            /// If the transaction is a freeze, the expected frozen/unfrozen outputs.
+            freeze_outputs,
+            inputs,
+            outputs,
+            time,
+            asset,
+            kind,
+            senders,
+            receivers,
+            fee_change: None,   // fee_change
+            asset_change: None, // asset_change
+            receipt: None,      // receipt
+        };
+        self.expiring_txns
+            .entry(timeout)
+            .or_insert_with(HashSet::default)
+            .insert(txn.uid().clone());
+        let mut editor = TransactionEditor::new(&mut self.store, txn);
+        editor.save()?;
+        Ok(editor)
+    }
+
+    /// Deletes an transaction from the store.
+    ///
+    /// Returns the deleted transaction.
+    pub fn delete(&mut self, code: &TransactionUID<L>) -> Result<Transaction<L>, KeystoreError<L>> {
+        Ok(self.store.delete(code)?)
+    }
+}
```

Comment:
In the spec we also have `delete_with_sk` for each secondary key, e.g. `delete_with_hash`. Not sure if this would ever be used. We can add it during the integration step if it turns out to be necessary

Reply:
I skipped adding these because I couldn't think of a reason we'd ever use them.  I am not sure we'd ever even use delete but I kept it in because it seems simple